pkg/operator/controller/ingress/controller.go

```diff
@@ -100,6 +100,16 @@ func New(mgr manager.Manager, config Config) (controller.Controller, error) {
 	if err != nil {
 		return nil, err
 	}
+	// Index IngressControllers over the default certificate name so that
+	// secretToIngressController can look up IngressControllers that
+	// reference the secret.
+	if err := operatorCache.IndexField(context.Background(), &operatorv1.IngressController{}, "ingressDefaultCertificateName", func(o client.Object) []string {
+		secret := operatorcontroller.RouterEffectiveDefaultCertificateSecretName(o.(*operatorv1.IngressController), operatorcontroller.DefaultOperandNamespace)
+		return []string{secret.Name}
+	}); err != nil {
+		return nil, fmt.Errorf("failed to create index for ingresscontroller: %v", err)
```

Comment:
```suggestion
		return nil, fmt.Errorf("failed to create index for ingresscontroller: %w", err)
```
But this isn't worth worrying about if you aren't making any other changes.

Reply:
Done.